pkg/splunk/enterprise/util.go

```diff
@@ -896,9 +931,14 @@ func handleAppRepoChanges(client splcommon.ControllerClient, cr splcommon.MetaOb
 			}
 		}
 
+		scope := getAppSrcScope(appFrameworkConfig, appSrc)
 		// 2.2 Check for any App changes(Ex. A new App source, a new App added/updated)
-		if AddOrUpdateAppSrcDeploymentInfoList(&appSrcDeploymentInfo, s3Response.Objects) {
-			appsModified = true
+		appsModified = AddOrUpdateAppSrcDeploymentInfoList(&appSrcDeploymentInfo, s3Response.Objects)
+
+		// if some apps were modified or added, and we have cluster scoped apps,
+		// then initialize the bundle push state
+		if appsModified && scope == enterpriseApi.ScopeCluster {
+			appDeployContext.BundlePushStatus.BudlePushStage = enterpriseApi.BundlePushNotStarted
```

Comment:
I feel this is redundant, as we are already preparing all the `cluster` scoped apps in the scheduler, and that is where we can set the stage to pending directly.